File: Lib/Telerik/Examples/RichTextEditor/StructuredContent/RecipeRangeStart.cs
using System;
using Telerik.WinForms.Documents.Model;

namespace RichTextEditor.StructuredContent
{
    public class RecipeRangeStart : AnnotationRangeStart
    {
        #region Properties

        [XamlSerializable]
        public string Name { get; set; }

        public RecipeRangeStart(string name)
        {
            this.Name = name;
        }

        public override bool SkipPositionBefore
        {
            get { return true; }
        }

        #endregion

        #region constructors

        public RecipeRangeStart()
        {
        }

        #endregion

        #region Methods

        protected override void CopyContentFromOverride(DocumentElement fromElement)
        {
            this.Name = ((RecipeRangeStart)fromElement).Name;
        }

        protected override DocumentElement CreateNewElementInstance()
        {
            return new RecipeRangeStart();
        }

        #endregion
    }
}

File: Lib/Telerik/Examples/RichTextEditor/StructuredContent/RecipeRangeStart.vb

Imports Telerik.WinForms.Documents.Model

Namespace RichTextEditor.StructuredContent
    Public Class RecipeRangeStart
        Inherits AnnotationRangeStart
#Region "Properties"

        <XamlSerializable> _
        Public Property Name() As String
            Get
                Return m_Name
            End Get
            Set(value As String)
                m_Name = Value
            End Set
        End Property
        Private m_Name As String

        Public Sub New(name As String)
            Me.Name = name
        End Sub

        Public Overrides ReadOnly Property SkipPositionBefore() As Boolean
            Get
                Return True
            End Get
        End Property

#End Region

#Region "constructors"

        Public Sub New()
        End Sub

#End Region

#Region "Methods"

        Protected Overrides Sub CopyContentFromOverride(fromElement As DocumentElement)
            Me.Name = DirectCast(fromElement, RecipeRangeStart).Name
        End Sub

        Protected Overrides Function CreateNewElementInstance() As DocumentElement
            Return New RecipeRangeStart()
        End Function

#End Region
    End Class
End Namespace
